### PR TITLE
feat: add `chainweaver suggest <flow>` static optimizer (#155)

### PR DESCRIPTION
**Pull request by dgenio**
## Summary

Adds `suggest_optimizations()` — a pure-static advisory pass that inspects a `Flow` and emits `Suggestion` objects with stable codes, plus a `chainweaver suggest <flow>` CLI subcommand for operator access.

Closes #155.

---

## Suggestion families

| Code | Slug | Requires |
|------|------|----------|
| **CW001** | `wasteful-passthrough` | — |
| **CW002** | `parallelizable-pair` | `--tools` |
| **CW003** | `dead-step` | `--tools` |
| **CW004** | `cacheable-step` | ≥ 2 `--trace` files |

---

### `chainweaver/analyzer.py`
- `Suggestion` Pydantic model with `code`, `title`, `step_index`, `tool_name`, `message`.
- `suggest_optimizations(flow, *, tools=None, traces=None)` public entry point.
- Four private helpers (`_suggest_wasteful_passthroughs`, `_suggest_parallelizable_pairs`, `_suggest_dead_steps`, `_suggest_cacheable_steps`).
- Empty-mapping (full-context passthrough) correctly treated as reading the tool's `input_schema` fields for CW002/CW003.

### `chainweaver/cli.py`
- `chainweaver suggest <flow>` command with `--tools`, `--trace`, `--format` options.
- Always exits 0 (advisory); exit 1 for malformed input, 2 for missing file.

### `chainweaver/__init__.py`
- Exports `Suggestion` and `suggest_optimizations` in `__all__`.

### `tests/test_cli_suggest.py`
- Programmatic tests for all four families + edge cases (DAGFlow, missing tools, single trace).
- CLI integration tests (table output, JSON output, `--tools` module loading, `--trace` loading, error exits).
- Regression tests for CW002/CW003 false positives on empty `input_mapping`.

---

## Validation

All four commands pass:
```
ruff check ✓ | ruff format --check ✓ | mypy ✓ | pytest 654 passed (94% coverage) ✓
```

### CODE REVIEW

File: chainweaver/analyzer.py [hunk 1]
```diff
@@ -259,3 +273,259 @@ def suggest_flows(
...
+                    step_index=idx,
+                    tool_name=a.tool_name,
+                    message=(
+                        f"Steps {idx} ('{a.tool_name}') and {idx + 1} ('{b.tool_name}') "
+                        "have disjoint output/input fields; they're DAG-eligible "
+                        "(promote to DAGFlow for parallel execution)."
+                    ),
+                )
+            )
+    return out
```
[the remark this answers is not in the document]
Comment (dgenio) [reply chainweaver/analyzer.py:408]: Agreed — fixed in e60b13c. CW002 now determines step B's actual reads based on `input_mapping`: empty mapping → uses `input_schema.model_fields` (full-context passthrough); non-empty → uses `_step_reads(b)` (mapped sources only). Added a regression test (`test_cw002_not_flagged_when_consumer_uses_empty_mapping`).

File: chainweaver/analyzer.py [hunk 2]
```diff
@@ -259,3 +273,259 @@ def suggest_flows(
...
+                    step_index=idx,
+                    tool_name=step.tool_name,
+                    message=(
+                        f"Step {idx} ('{step.tool_name}') outputs "
+                        f"{sorted(produced)} but no downstream step reads them. "
+                        "Remove the step or wire it into a downstream input_mapping."
+                    ),
+                )
+            )
+    return out
```
[the remark this answers is not in the document]
Comment (dgenio) [reply chainweaver/analyzer.py:462]: Valid catch — fixed in e60b13c. `_suggest_dead_steps` now treats downstream steps with empty `input_mapping` as implicitly reading their declared `input_schema` fields (since the executor passes the full context and Pydantic validates against the schema). Added a regression test (`test_cw003_not_flagged_when_downstream_uses_empty_mapping`).

File: chainweaver/analyzer.py [hunk 3]
```diff
@@ -259,3 +273,259 @@ def suggest_flows(
...
+                title=SUGGESTION_CODES["CW001"],
+                step_index=idx,
+                tool_name=step.tool_name,
+                message=(
+                    f"Step {idx} ('{step.tool_name}') uses an empty input_mapping. "
+                    "Consider an explicit mapping so the data flow is visible to readers."
+                ),
+            )
+        )
+    return out
```
[the remark this answers is not in the document]
Comment (dgenio) [reply chainweaver/analyzer.py:346]: Good catch on the spec drift (lesson #1). Fixed the docstring in e60b13c to accurately describe the implementation: it flags every step with an empty `input_mapping` regardless of whether the tool declares input fields (tool schemas aren't available in this function). The current behavior is pragmatically sound — flagging is advisory, and zero-input tools with empty mappings are rare.
